### ISSUE TITLE
[Feature Request] File/link association

### ISSUE THREAD

**Issue by toomyzoom** [husudosu/mpv-remote-app#2]
Hi, thank you for your remote app. It's a much improved version compared to mcastorina's version.

I have a request. Can you add file association to open different filetype/link with your android remote app?
I have some that come to mind:
- Audio file/stream
- Video file/stream
- Torrent/magnet link
- Zip file/link
- HTML link (sometime the link is not obvious with file extension)

I have other apps open the video file url but your remote app doesn't have link/video file association so it's kind of a pain to find way to copy the link to internal dialog.

Thank you.

**Comment by husudosu** [comment]
Hey! Sorry for taking time, I've implemented the requested feature, you can download the 1.0.3 version from here:
- [Please update backend first](https://github.com/husudosu/mpv-remote-node/releases/tag/1.0.3)
- [If it's updated update the app also](https://github.com/husudosu/mpv-remote-app/releases/tag/1.0.3)

The backend also available on NPM registry from now on, so I recommend to remove old package and reinstall the whole thing with using:
```bash
npm uninstall -g mpv-remote
npm install -g mpv-remote
mpv-remote # Follow instructions
```

I couldn't test webtorrent, because I can't get it work on my Windows machine, but it should work. The backend API uses loadfile method of MPV.
With Youtube app works fine for me.
Please test it, and let me know if it's working or not!

Have a nice day!

**Comment by toomyzoom** [comment]
Thanks. That was quick.

Tried the 1.0.3. There are 2 issues:

- The app is not associated with media file/link. It's only available in share menu.
- Because the app only available in share menu, I have to use "Share & Open" app in playstore to open the link and share it to mpvremote app. This caused another issue. Seem like the url in the intent is not properly formatted as string. The server threw this error:
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote] { flag: 'append-play' }
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string or an instance of Buffer or URL. Received undefined
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at Object.lstatSync (node:fs:1507:10)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at /usr/lib/node_modules/mpv-remote/remoteServer.js:465:10
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at Layer.handle [as handle_request] (/usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/layer.js:95:5)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at next (/usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/route.js:137:13)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at Route.dispatch (/usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/route.js:112:3)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at Layer.handle [as handle_request] (/usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/layer.js:95:5)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at /usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/index.js:281:22
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at Function.process_params (/usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/index.js:335:12)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at next (/usr/lib/node_modules/mpv-remote/node_modules/express/lib/router/index.js:275:10)
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]     at /usr/lib/node_modules/mpv-remote/node_modules/body-parser/lib/read.js:130:5 {
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote]   code: 'ERR_INVALID_ARG_TYPE'
Nov 17 23:06:30 mpv xinit[6666]: [mpvremote] }

**Comment by husudosu** [comment]
I'll look into it. Thanks for the feedback.
Maybe I misunderstood something. You also want to stream video or audio  files from your phone?
I'm not sure that's implementable with ionic. Maybe with a native capacitor or cordova plugin.
Don't know if mpv loadfile can handle pure data stream (Don't have that much experience with MPV internals) .

**Comment by toomyzoom** [comment]
Sorry, I was not being clear about my request.
I didn't mean to stream from phone. That doesn't make sense because it will need something like a local http server on the phone.

Let put out an example for my use case. I use an app (Aniyomi) that grab url from streaming site. It has it own native player but I can set it to use external player. Most of the links for its output is either m3u8 and mp4.  Seem like when it creates the intent for those, android specifies only the apps associate with those file type can be in "Open with" menu. I assume because the intent filter is "android.intent.action.SEND" and the mimetype is plaintext. In the media player app, they have more extensive intent filter and they are under "android.intent.action.VIEW", which trigger the open with menu
https://github.com/mstorsjo/vlc-android/blob/master/vlc-android/AndroidManifest.xml
https://github.com/mpv-android/mpv-android/blob/master/app/src/main/AndroidManifest.xml

**Comment by husudosu** [comment]
Thanks for the explanation. :) 
I'll start working on it.

**Comment by husudosu** [comment]
Is this the behaviour you want?
https://www.youtube.com/watch?v=3XGeEmU7yc0

It still need some work. I've to reimplement Youtube intent handling and maybe also give webtorrent an other try (now have access to proper Linux machine) .
I can't guarantee releasing a new version this week.

**Comment by husudosu** [comment]
Implemented, 1.0.4 version available for a while, I'm closing this issue. If it's not working for you, re-open the issue!
